server/bell_oauth_profile.js

```diff
@@ -0,0 +1,52 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+'use strict';
+
+var log = require('./logger')('server.bell.profile');
+var db = require('./db/db');
+var config = require('./config');
+
+// this is the custom provider profile function used by Bell to allow us
+// to convert oauth tokens into profile information.
+function profile (credentials, params, get, profileCb) {
```

Comment:
The general server-side oauth flow is this:
- user surfs to `/auth/login`, which redirects to `/auth/complete` (because Bell expects only one endpoint for the whole oauth dance, I dunno why)
- bell then takes over, generates an oauth nonce, and redirects the user to fxa oauth endpoint
- the user logs in, fxa redirects back to `/auth/complete`
- bell validates the nonce
- bell then calls this `profile` function, which is supposed to take a valid oauth token, and ask some other endpoint for profile information (the outgoing request is made by the `get` function at line 16)
- once `get` returns profile info, we see if the user exists, and either create or update the user db,
- then call profileCb, which passes the fxa ID back to the `/auth/complete` route handler
- the `/auth/complete` handler inserts that ID into a session cookie.

Feel free to ask if part of this seems unclear.